src/univariates.jl

```diff
@@ -367,8 +367,27 @@ logccdf(d::UnivariateDistribution, x::Real) = log(ccdf(d, x))
     quantile(d::UnivariateDistribution, q::Real)
 
 Evaluate the inverse cumulative distribution function at `q`.
+Use `quantile` to calculate quartiles, medians and confidence intervals.
```

Comment:
One shouldn't calculate the median with `quantile`. IMO confidence intervals are not properties of distributions and one usually just wants to use `confint` which is not part of Distributions.

Reply:
>One shouldn't calculate the median with quantile.

Then what is this, from https://github.com/JuliaStats/Distributions.jl/blob/master/src/univariates.jl:
![image](https://user-images.githubusercontent.com/61620837/149916979-26c67ced-c336-4828-8767-ff6d21bd5d77.png)

> IMO confidence intervals are not properties of distributions...

No they are not a property of the distribution, agreed. Which is why it could be nice with an example of what the function can be used for. To make realizing the potential of the function easier.

> ... and one usually just wants to use confint which is not part of Distributions.

Right. But I found manual computation of confidence intervals very useful in learning about them in a statistics course. And there are LOADS of people taking statistics courses. Which makes me thing that catering a little bit to that audience is justified. Also, there might be other cases where manual computation of confidence intervals would be useful. I am simply suggesting to show how it is done, _in case_ someone needs it. Which I know I did. And which I am confident that someone will need again.

There have been several posts (like these from [discourse](https://discourse.julialang.org/t/package-for-confidence-intervals/33902) and [stackoverflow](https://stackoverflow.com/questions/20425473/calculate-confidence-interval-in-julia)) asking for how to compute confidence intervals. I also had that question, without making a post, suggesting to me that many people have had this question. I think that there is clearly some degree of need for this documentation, and I don't really see the downside...

Reply:
> Then what is this

Well, a "fallback" is something you only do if you don't have anything better/specific.

Reply:
Snap, my bad. Thanks for the patience.

Alright, then I'll remove the suggestion to calculate medians this way.

Reply:
> Then what is this, from https://github.com/JuliaStats/Distributions.jl/blob/master/src/univariates.jl:

As is noted in the docstring, this is a fallback. Users who want `median` should call `median` directly, since there might be a more efficient overload for that distribution. e.g. for `Normal`, the mean is the median and can be directly returned without calling the inverse error function as `quantile` does.

I should note that it's normal to just define a quantile function as the inverse CDF as is done here. e.g. see [documentation for `scipy.stats.norm.ppf`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html), or the [docstring for `qnorm` in R](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/Normal.html). The quantile function has many uses, and it seems unnecessary to inform the user of them in my opinion.

Reply:
I agree with your points @sethaxen. Do you see a solution where users would still get a result and some pointers when searching for `confidence interval` in the documentation? Because as a user, I felt that was lacking, and that is the underlying problem I am trying to fix.

Reply:
I don't think it's lacking for the same reason I don't think it's a problem that in the docstring for `logpdf` we don't say that it could, for example, be used to sample from a distribution using Markov Chain Monte Carlo. These are very basic ingredient functions, and trying to document what they can be used for is an endless task. So IMO, if a user doesn't know how a CI relates to a quantile function, then they shouldn't be trying to compute a CI.

And FWIW, if you search "confidence interval scipy", for example, your hits will all be from StackOverflow detailing how to use SciPy's ingredient functions, not from the SciPy docs itself.

Reply:
Fair points. So perhaps posing the question on Discourse, and answering it myself, is the way to go...

Reply:
Still let’s add examples, that’s always good, even an probability interval if we want.

Reply:
Thanks to you both for your conciderations ^_^

Reply:
> Fair points. So perhaps posing the question on Discourse, and answering it myself, is the way to go...

I ended up making the following post, for googlability:
https://discourse.julialang.org/t/how-to-compute-calculate-a-confidence-interval-ci/74794/2